### PR TITLE
ZHA channel changing dialog and settings redesign

### PR DESCRIPTION
**Pull request by puddly**
## Proposed change 

Redesign the ZHA network settings section and incorporate a button and dialog to change the current channel:

### Before
before

### After
after
</div>

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        <p>
+          <ha-select
+            .label=${this.hass.localize(
+              "ui.panel.config.zha.change_channel_dialog.new_channel"
+            )}
+            fixedMenuPosition
+            naturalMenuWidth
+            @selected=${this._newChannelChosen}
+            @closed=${stopPropagation}
+            value=${String(this._params.currentChannel)}
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts:87]:
```suggestion
            .value=${String(this._params.currentChannel)}
```

Comment (bramkragten) [reply src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts:87]: The value should be the selected value, so should be `this._newChannel || this._params.currentChannel`

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+@customElement("dialog-zha-migrate-channel")
+class DialogZHAMigrateChannel extends LitElement implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _migrationInProgress = false;
+
+  @state() private _params?: ZHAMigrateChannelDialogParams;
+
+  private _newChannel: "auto" | number = "auto";
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts:43]: `_newChannel` should be initialized with `params.currentChannel` I think, and not with `auto`?

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+  public async showDialog(
+    params: ZHAMigrateChannelDialogParams
+  ): Promise<void> {
+    this._params = params;
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts:52]:
```suggestion
    this._params = undefined;
    this._newChannel = undefined;
```

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts (hunk 2 above)
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts:43]:
```suggestion
  @state() private _newChannel?: "auto" | number;
```

File: src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts [hunk 4]
```diff
@@ -126,30 +134,54 @@ class ZHAConfigDashboard extends LitElement {
...
-            ? html`<div class="card-content network-settings">
-                <div>
-                  <strong>PAN ID:</strong>
-                  ${this._networkSettings.settings.network_info.pan_id}
-                </div>
-                <div>
-                  <strong>Extended PAN ID:</strong>
-                  ${this._networkSettings.settings.network_info.extended_pan_id}
+            ? html`<div class="card-content">
+                <div class="item">
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts:138]: Can we use `ha-settings-row` here?

**Review by bramkragten — APPROVED** [review]

**Comment by Hedda** [comment]
@frenck please consider adding this to Home Assistant 2023.6 release notes ([it is not mentioned in beta release notes](https://rc.home-assistant.io/changelogs/core-2023.6) as of yet).

This experimental feature allows ZHA users to perform Zigbee channel migration directly from the UI for all supported radio types without the need to re-pair all devices or have any entities changed. ZNP (Texas Instruments) based devices currently has the best implementation of Zigbee channel changing as it allows the Zigbee Coordinator to change Zigbee channels after having a few chances to send out the broadcast to all the devices. Other Zigbee Coordinator adapters (e.g. Silicon Labs EZSP and Dresden Elektronik's deconz based radios) instead migrate/change the Zigbee channel almost immediately, giving Zigbee End Devices (e.i. battery-powered devices) a smaller window of receiving the channel change request broadcast, however, such devices still have a chance of finding the network again if they detect that they have been orphaned. Modern mains-powered devices should normally all change channels without any problems. After waiting a while for all devices to change channels automatically the general recommendation is then to perform a power cycle of devices that did not automatically change channels (usually simply by removing and replacing the battery) before just re-pairing those devices in the ZHA integration. 

References:

https://github.com/home-assistant/frontend/pull/16381

https://github.com/home-assistant/core/pull/92076

https://github.com/home-assistant/core/pull/91969

https://github.com/zigpy/zigpy/pull/1190

**Pull request merged**